src/main/java/seedu/fast/model/tag/Tag.java

```diff
@@ -32,6 +34,23 @@ public static boolean isValidTagName(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns true if a given string is a valid tag name.
+     */
+    public static boolean isValidTagTerm(String test) {
+        return test.matches(VALIDATION_REGEX) || test.matches(PriorityTag.PRIORITY_VALIDATION_REGEX);
+    }
+
+    /**
+     * Create either an instance of a Tag or a PriorityTag depending on the input term.
+     */
+    public static Tag createTag(String term) {
+        if (term.matches(PriorityTag.PRIORITY_VALIDATION_REGEX)) {
+            return new PriorityTag(term);
+        }
+        return new Tag(term);
+    }
```

Comment:
Does this mean that the users can only either create a normal tag or a priority tag? Since the current implementation of the tag using edit clears current tags and remakes them.

Reply:
Yes, users can either create a normal tag or a priority tag. But with the current implementation of edit, it is possible to create multiple tags for the same contact with a single edit command. 
For example:
```
edit 2 t/ pr/medium t/ fat
//gives the second indexed contact a MediumPriority Tag and a fat Tag
```